viewer/esProxy.js

```diff
@@ -0,0 +1,280 @@
+  }
+}
+
+// GET calls we can match exactly
+let getExact = {
+  '/': 1,
+  '/_cat/health': 1,
+  '/_cluster/health': 1,
+  '/_nodes/stats/jvm,process,fs,os,indices,thread_pool': 1
+};
+getExact[`/_template/${prefix}sessions2_template`] = 1;
+getExact[`/${prefix}sessions2-*/_alias`] = 1;
+getExact[`/${prefix}stats/_stats`] = 1;
+getExact[`/${prefix}users/_stats`] = 1;
+getExact[`/${prefix}users/_count`] = 1;
+getExact[`/${prefix}sequence/_stats`] = 1;
+getExact[`/${prefix}dstats/_stats`] = 1;
+getExact[`/${prefix}files/_stats`] = 1;
+getExact[`/${prefix}fields/_search`] = 1;
+
+// POST calls we can match exactly
+let postExact = {
+};
+postExact[`/${prefix}stats/_search`] = 1;
+postExact[`/${prefix}fields/_search`] = 1;
+postExact[`/${prefix}lookups/_search`] = 1;
+
+// ============================================================================
+// Auth
+// ===========================================================================
+
+app.use((req, res, next) => {
+  let credentials = basicAuth(req);
+  if (!credentials) {
+    return res.set('WWW-Authenticate', 'Basic').status(401).send();
+  }
+
+  if (!sensors[credentials.name]) {
+    console.log(`Unknown sensor ${credentials.name}`);
+    return res.set('WWW-Authenticate', 'Basic').status(401).send();
+  }
+
+  if (sensors[credentials.name].pass !== undefined && sensors[credentials.name].pass !== credentials.pass) {
+    console.log(`Incorrect password for ${credentials.name}`);
+    return res.set('WWW-Authenticate', 'Basic').status(401).send();
+  }
+
+  req.sensor = sensors[credentials.name];
+  if (!sensors[credentials.name].ip) {
+    return next();
+  }
+
+  let ip = req.connection.remoteAddress;
+  if (ip.startsWith('::ffff:')) {
+    ip = ip.substring(7);
+  }
+
+  if (!req.sensor.ip.includes(ip)) {
+    console.log(`Incorrect source ip ${ip} for ${credentials.name}`);
+    return res.set('WWW-Authenticate', 'Basic').status(401).send();
+  }
+
+  return next();
+});
+
+// ============================================================================
+// Proxy code to real ES
+// ===========================================================================
+
+// Save the post body
+
+function hasBody (req) {
+  var encoding = 'transfer-encoding' in req.headers;
+  var length = 'content-length' in req.headers && req.headers['content-length'] !== '0';
+  return encoding || length;
+}
+
+function saveBody (req, res, next) {
+  if (req._body) { return next(); }
+  req.body = req.body || {};
+
+  if (!hasBody(req)) { return next(); }
+
+  // flag as parsed
+  req._body = true;
+
+  // parse
+  var buf = Buffer.alloc(parseInt(req.headers['content-length'] || '1024'));
+  let pos = 0;
+  req.on('data', (chunk) => { chunk.copy(buf, pos); pos += chunk.length; });
+  req.on('end', () => {
+    req.body = buf;
+    next();
+  });
+}
+
+// Proxy
+
+function doProxy (req, res, cb) {
+  let result = '';
+  let url = elasticsearch + req.url;
+  console.log('URL', url);
+  var info = URL.parse(url);
```

Comment:
Any reason you're using `var` in some places and `let` in others??

Reply:
copy paste, fixing